### PR TITLE
Backport more root6 cleanups

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
-1

merge conflicts and compilation errors should be resolved.

**Comment by wmtan** [comment]
@ktf  I am doing similar backports.  Can you let me know what other backports you are doing so we don't duplicate effort, or, even worse, collide.

**Comment by wmtan** [comment]
-1
I know I am not an L2 for this, but the change to TrackingTools/TrajectoryState/BuildFile.xml
needs to be removed from this.  The ROOT6 version is being changed in my pull request to tha superior ROOT 5 version.  This will undo the change.
The changes tio the other three files are fine.

File: TrackingTools/TrajectoryState/BuildFile.xml [hunk 1]
```diff
@@ -6,11 +6,11 @@
 <use   name="DataFormats/TrackReco"/>
 <use   name="DataFormats/TrajectoryState"/>
 <use   name="DataFormats/BeamSpot"/>
+<use   name="MagneticField/Engine"/>
 <use   name="TrackingTools/AnalyticalJacobians"/>
 <use   name="TrackingTools/TrajectoryParametrization"/>
-<use   name="MagneticField/Engine"/>
 
-<use   name="vdt_headers"/>
+<use   name="vdt"/>
```
Comment (wmtan) [inline TrackingTools/TrajectoryState/BuildFile.xml]: Do not make this change.  The 7_4_X dependency is the correct one, and I already have a PR that changes the 7_4_ROOT6_X version of this file to the 7_4_X version

Comment (ktf) [inline TrackingTools/TrajectoryState/BuildFile.xml]: This was done a couple of days ago. Please go ahead with your changes.

Comment (wmtan) [inline TrackingTools/TrajectoryState/BuildFile.xml]: Giulio, Thanks for the prompt reply. But I need to know which of these two options you prefer:
1) This PR will be closed, and I will submit the changes to these other three files in my PR.s
2) This PR will remain open, with the changes to this one file removed.
Thank you in either case.
Bill

**Comment by ktf** [comment]
my laptop broke so I'm without computer until Monday. I'm closing this.

**Pull request closed without merging**